src/PowerShellEditorServices/Utility/Utils.cs

```diff
@@ -17,5 +18,38 @@ internal static class Utils
         /// True if we are running on .NET Core, false otherwise.
         /// </summary>
         public static bool IsNetCore { get; } = RuntimeInformation.FrameworkDescription.StartsWith(".NET Core", StringComparison.Ordinal);
+
+        /// <summary>
+        /// Get's the Version of PowerShell being used.
+        /// </summary>
+        public static Version PSVersion { get; } = PowerShellReflectionUtils.PSVersion;
+
+        /// <summary>
+        /// True if we are running in Windows PowerShell, false otherwise.
+        /// </summary>
+        public static bool IsPS5 { get; } = PSVersion.Major == 5;
+
+        /// <summary>
+        /// True if we are running in PowerShell Core 6, false otherwise.
+        /// </summary>
+        public static bool IsPS6 { get; } = PSVersion.Major == 6;
+
+        /// <summary>
+        /// True if we are running in PowerShell 7, false otherwise.
+        /// </summary>
+        public static bool IsPS7 { get; } = PSVersion.Major == 7;
```

Comment:
Consider future proofing this somewhat by making it `IsPS7OrGreater`.  It doesn't really matter since we aren't using it, but could save us some trouble if it gets used somewhere.

Reply:
I’d like to keep this as IsPS7. I get the idea that PS8 will maybe be on .NET 5 which might have differences… not too mention that will be years out. Plus, if PS7 makes it back into Windows in a few years, maybe there will be a day where we drop `IsPS5` so this file is certainly “living” anyway.